src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3282,6 +3289,26 @@ TransactionStateMachine(WorkerSession *session)
 					break;
 				}
 
+				/* if this is a multi-query task, send the next query */
+				if (placementExecution->queryIndex < task->queryCount)
+				{
+					ClearResults(connection, false);
+
+					bool querySent = SendNextQuery(placementExecution, session);
+					if (!querySent)
+					{
+						/* no need to continue, connection is lost */
+						Assert(session->connection->connectionState ==
+							   MULTI_CONNECTION_LOST);
+
+						return;
+					}
+
+					transaction->transactionState = REMOTE_TRANS_SENT_COMMAND;
```

Comment:
do we need to update wait flags here? we need to be listening for WRITEABLE

Reply:
that's correct. done.